pkg/kubectl/cmd/cmd_test.go

```diff
@@ -233,9 +235,14 @@ func NewAPIFactory() (*cmdutil.Factory, *testFactory, runtime.Codec) {
 		ClientConfig: func() (*client.Config, error) {
 			return t.ClientConfig, t.Err
 		},
-		Generator: func(name string) (kubectl.Generator, bool) {
-			generator, ok := generators[name]
-			return generator, ok
+		ExposeGenerators: func() map[string]kubectl.Generator {
```

Comment:
Agreed, I was worried about it. I started with a 2d map and I think I prefer it but @deads2k had some objections. It's possible I didn't understand what he was trying to say though:)